### PR TITLE
Editable bin/setup file template

### PR DESCRIPTION
**Pull request by mstanielewicz**
Updates bin/setup script to use config files and add interactive user input.

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -8,42 +8,62 @@ def say(*args)
...
-  copy_sample :database
-  copy_sample :secrets
+  header "Copying and configuring sample files"
+  # Add copying and configuring sample files here:
+  # (Example configures database.yml)
+  # You can use gsub_file to add other values to any file.
+  # For example: gsub_file("config/secrets.yml", /SECRET_KEY_BASE:.*\n/, "SECRET_KEY_BASE: #{SecureRandom.hex(64)}")
+  copy_sample "config/database.yml"
+  say "\nWhat is your database username? ", :yellow
+  username = gets
```
Comment (lubieniebieski) [inline bin/setup]: we can use more goodness of `Thor` here - there's a method calles `ask`

File: bin/setup [hunk 2]
```diff
@@ -1,49 +1,105 @@
...
-  real_name = "config/#{name}.yml"
-  if File.exist? real_name
-    say "#{real_name} already exists"
-  else
-    system "cp #{real_name}.sample #{real_name}"
+def set_nil_paths(hash, current_path=[])
+  hash.each do |key, value|
+    if value.kind_of? Hash
+      set_nil_paths(value, current_path.clone.push(key))
+    elsif value.nil?
```
Comment (lubieniebieski) [inline bin/setup]: maybe we should always ask about it? I mean, maybe I want to setup this project few months later and there are some new keys here and I would like to see them?

something like:

```
Do you want to change secret_token value? (currently: "foo bar baz") 
```

(press `S` to skip) or something like that. What do you think @mstanielewicz ?

File: bin/setup [hunk 3]
```diff
@@ -1,49 +1,105 @@
...
+  # You can use gsub_file to add other values to any file.
+  # For example: gsub_file("config/secrets.yml", /SECRET_KEY_BASE:.*\n/, "SECRET_KEY_BASE: #{SecureRandom.hex(64)}")
+  if copy_sample "config/database.yml"
+    gsub_file("config/database.yml", /username:.*\n/, "username: #{ ask("\nWhat is your database username? ", :blue) }\n")
+    gsub_file("config/database.yml", /password:.*\n/, "password: #{ ask("\nWhat is your database password? ", :blue) }\n")
+  end
+
+  # For config file in YAML format:
+  if ::File.exists? "#{CONFIG_FILE}.sample"
+    @config_hash = YAML.load ::File.read("#{CONFIG_FILE}.sample")
```
Comment (lubieniebieski) [inline bin/setup]: do you need an instance variable `@` here?

File: bin/setup [hunk 4]
```diff
@@ -1,49 +1,105 @@
...
+  File.write(rel_path, content)
 end
 
-def copy_sample(name)
-  real_name = "config/#{name}.yml"
-  if File.exist? real_name
-    say "#{real_name} already exists"
-  else
-    system "cp #{real_name}.sample #{real_name}"
+def set_nil_paths(hash, current_path=[])
```
Comment (lubieniebieski) [inline bin/setup]: how about renaming it to `set_local_config_values`?

File: bin/setup [hunk 5]
```diff
@@ -1,49 +1,105 @@
...
 
-# path to your application root.
+# Path to your application root:
 APP_ROOT = Pathname.new File.expand_path('../../',  __FILE__)
 
-Dir.chdir APP_ROOT do
-  # This script is a starting point to setup your application.
-  # Add necessary setup steps to this file:
+# Relative path to your config file:
+CONFIG_FILE = "config/sec_config.yml"
```
Comment (lubieniebieski) [inline bin/setup]: we use `secrets.yml` in this project

File: bin/setup [hunk 6]
```diff
@@ -1,49 +1,165 @@
...
+  # Add copying and configuring sample files here:
+  # (Example configures database.yml)
+  # You can use gsub_file to add other values to any file.
+  # For example: gsub_file("config/secrets.yml", /SECRET_KEY_BASE:.*\n/, "SECRET_KEY_BASE: #{SecureRandom.hex(64)}")
+
+  # Uncomment if you need to set database username and password
+  # if copy_sample "config/database.yml"
+  #   gsub_file("config/database.yml", /username:.*\n/, "username: #{ ask("\nWhat is your database username? ", :blue) }\n")
+  #   gsub_file("config/database.yml", /password:.*\n/, "password: #{ ask("\nWhat is your database password? ", :blue) }\n")
+  # end
```
Comment (jniechcial) [inline bin/setup]: These comments may be removed from here - these multiple possibilites are described on Confluence.

File: bin/setup [hunk 7]
```diff
@@ -1,49 +1,165 @@
...
+  # For .env file:
+  # if File.exists? ".env.sample"
+  #   File.write(".env", File.read(".env.sample")
+  #                          .split("\n")
+  #                          .map{|key| key.split("=")}
+  #                          .each{|key| key.push(ask("\nSet #{key[0]}: ")) unless key[1]}
+  #                          .map{|key| key.join("=")}
+  #                          .join("\n")
+  #   )
+  # end
```
Comment (jniechcial) [inline bin/setup]: Comments as formerly

File: bin/setup [hunk 8]
```diff
@@ -1,49 +1,165 @@
...
 
   header "Preparing database"
-  system "bin/rake db:setup"
+  # Simple database setup here:
+  # You can change it to `system "bin/rake db:setup"` or add additional commands
+  # like `system "bin/rake db:test:prepare"`
```
Comment (jniechcial) [inline bin/setup]: Comments as formerly

**Comment by jniechcial** [comment]
@mstanielewicz Can you run Rubocop on this PR?

**Pull request merged**